app/scripts/controllers/incoming-transactions.js

```diff
@@ -234,3 +265,20 @@ class IncomingTransactionsController {
 }
 
 module.exports = IncomingTransactionsController
+
+function pairwise (fn) {
+  let first = true
+  let cache
+  return (value) => {
+    try {
+      if (first) {
+        first = false
+        return fn(value, value)
```

Comment:
Returning `value` as the previous value makes sense in the context of this listener. But if you wanted this to be a more generally useful utility function, I think it'd make sense to use `null` as the previous value instead. It would be easy enough to handle a null previous value in the handlers.